core/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/MockDiskUsagesTests.java

```diff
@@ -137,6 +138,28 @@ public void run() {
         assertThat("node1 has 5 shards", nodesToShardCount.get(realNodeNames.get(0)), equalTo(5));
         assertThat("node2 has 5 shards", nodesToShardCount.get(realNodeNames.get(1)), equalTo(5));
         assertThat("node3 has 0 shards", nodesToShardCount.get(realNodeNames.get(2)), equalTo(0));
+
+        // Update the disk usages so one node is now back under the high watermark
+        cis.setN1Usage(realNodeNames.get(0), new DiskUsage(nodes.get(0), "n1", 100, 50));
+        cis.setN2Usage(realNodeNames.get(1), new DiskUsage(nodes.get(1), "n2", 100, 50));
+        cis.setN3Usage(realNodeNames.get(2), new DiskUsage(nodes.get(2), "n3", 100, 50)); // node3 has free space now
+
+        // Cluster info gathering interval is 2 seconds, give reroute 2 seconds to kick in
+        Thread.sleep(4000);
```

Comment:
Can you inject some clock interface rather than use System.nanotime and sleep? Or crank the reroute interval down to 0? Something like guava's Ticker.